clang/lib/CIR/Dialect/IR/CIRTypes.cpp

```diff
@@ -615,6 +615,244 @@ IntType::verify(llvm::function_ref<mlir::InFlightDiagnostic()> emitError,
   return mlir::success();
 }
 
+//===----------------------------------------------------------------------===//
+// FloatType Definitions
+//===----------------------------------------------------------------------===//
+
+Type cir::FloatType::parse(mlir::AsmParser &parser) {
+  auto *ctx = parser.getBuilder().getContext();
+  Format format;
+
+  if (parser.parseLess())
+    return {};
+
+  // Fetch floating-point format.
+  llvm::StringRef formatStr;
+  if (parser.parseKeyword(&formatStr))
+    return {};
+  if (formatStr == "bin16")
```

Comment:
Please remove all logic that is not relevant to this PR.

Reply:
Updated.